File: medium/198. House Robber/Solution.java
class Solution {
   public int rob(int[] nums) {
        if(nums.length == 1) return nums[0];

        int maxFirst = 0;
        int maxSecond = 0;
        int current = 0;

        for(int num : nums) {
            current = Math.max(maxSecond, maxFirst + num);
            maxFirst = maxSecond;
            maxSecond = current;
        }

        return current;
    }
}

File: medium/198. House Robber/Solution.kt
class Solution {
    fun rob(nums: IntArray): Int {
        if (nums.size == 1) return nums[0]

        var maxFirst = 0
        var maxSecond = 0

        for (num in nums) {
            val cur = maxSecond.coerceAtLeast(maxFirst + num)
            maxFirst = maxSecond
            maxSecond = cur
        }

        return maxSecond
    }
}